src/main/java/appeng/fluids/client/gui/GuiFluidInterface.java

```diff
@@ -56,17 +57,26 @@ public void initGui()
 	{
 		super.initGui();
 
+		final IAEFluidTank configFluids = this.host.getDualityFluidInterface().getConfig();
+		final IAEFluidTank fluidTank = this.host.getDualityFluidInterface().getTanks();
+
 		for( int i = 0; i < DualityFluidInterface.NUMBER_OF_TANKS; ++i )
 		{
-			final AEFluidTank fluidTank = this.host.getDualityFluidInterface().getTank( i );
-			final GuiFluidTank guiTank = new GuiFluidTank( ID_BUTTON_TANK + i, fluidTank, this.getGuiLeft() + 35 + 18 * i, this.getGuiTop() + 53, 16, 68 );
+			final GuiFluidTank guiTank = new GuiFluidTank( fluidTank, i, DualityFluidInterface.NUMBER_OF_TANKS + i, this.getGuiLeft() + 35 + 18 * i, this
+					.getGuiTop() + 53, 16, 68 );
 			this.buttonList.add( guiTank );
+			this.guiSlots.add( new GuiFluidSlot( configFluids, i, i, 35 + 18 * i, 35 ) );
 		}
 
 		this.priority = new GuiTabButton( this.getGuiLeft() + 154, this.getGuiTop(), 2 + 4 * 16, GuiText.Priority.getLocal(), this.itemRender );
 		this.buttonList.add( this.priority );
 	}
 
+	@Override
+	protected void addButtons()
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1186)